### ISSUE TITLE
Increasingly the replacement pastes recent clipboard content

### ISSUE THREAD

**Issue by ReaderGuy42** [federico-terzi/espanso#1670]
**Describe the bug**
A clear and concise description of what the bug is
over the past week or so, I've been getting the bug where instead of the correct replacement, a recently copied textbit from the clipboard is pasted. Most frequently I've noticed this in the case of multi-line expansion texts.

**To Reproduce**
1. trigger Espanso
2. sometimes it pastes text from clipboard

**Expected behavior**

regular Espanso output

**Setup information**
- OS: Linux Mint 21.2
- Version: 2.1.8

**Comment by xeruf** [comment]
yep, multi-line output - example is to use the output of the `env` shell command which reliably pastes the wrong text - but when I enable debug for it it prints the correct content in the logs

Manjaro Sway on Wayland here

**Comment by bdooleyatvgsidotcom** [comment]
I have also experienced this same issue.

Windows 11 24H2 (OS Build 26100.4652)
Espanso v2.2.1

**Comment by smeech** [comment]
Is `restore_clipboard_delay` from [options](https://espanso.org/docs/configuration/options/#options-reference) any help?

@xeruf, on Wayland, it can be about specifying the `keyboard_layout:` parameters correctly.

**Comment by bdooleyatvgsidotcom** [comment]
> Is `restore_clipboard_delay` from [options](https://espanso.org/docs/configuration/options/#options-reference) any help?
> 
> [@xeruf](https://github.com/xeruf), on Wayland, it can be about specifying the `keyboard_layout:` parameters correctly.

I will try changing 'restore_clipboard_delay'.  Thank you.
